### PR TITLE
[REVIEW] Gtest target names on cmake build

### PR DESCRIPTION
**Pull request by dantegd**

### CODE REVIEW

**Comment by GPUtester** [comment]
Please update the changelog in order to start CI tests.

View the gpuCI docs [here](https://docs.rapids.ai/gpuci#open-pr-or-update-pr).

File: cpp/cmake/Dependencies.cmake [hunk 1]
```diff
@@ -36,15 +36,15 @@ if(BUILD_GTEST)
...
-  add_dependencies(gtest_mainlib googletest)
+  add_dependencies(GTest::GTest googletest)
+  add_dependencies(GTest::Main googletest)
 
 else()
 
   find_package(GTest REQUIRED)
 
-endif(BUILD_TEST)
+endif(BUILD_GTEST)
```
Comment (hcho3) [inline cpp/cmake/Dependencies.cmake:50]: In latest CMake, it is no longer necessary to write out the condition expression in `endif()`.

Comment (hcho3) [reply cpp/cmake/Dependencies.cmake:50]: "Per legacy, the else() and endif() commands admit an optional <condition> argument. If used, it must be a verbatim repeat of the argument of the opening if command." https://cmake.org/cmake/help/v3.14/command/if.html

Comment (JohnZed) [reply cpp/cmake/Dependencies.cmake:50]: Wow, I hadn't heard... pretty cool!

**Pull request merged**